main.go

```diff
@@ -4,54 +4,67 @@ import (
 	"flag"
 	"log"
 	"os"
+	"strconv"
 
 	"github.com/docker/go-plugins-helpers/volume"
 )
 
-const quobyteID string = "quobyte"
+const (
+	quobyteID string = "quobyte"
+)
 
 var (
 	version  string
 	revision string
 )
 
 func main() {
-	quobyteMountPath := flag.String("path", "/run/docker/quobyte/mnt", "Path where Quobyte is mounted on the host")
-	quobyteMountOptions := flag.String("options", "-o user_xattr", "Fuse options to be used when Quobyte is mounted")
-
-	quobyteUser := flag.String("user", "root", "User to connect to the Quobyte API server")
-	quobytePassword := flag.String("password", "quobyte", "Password for the user to connect to the Quobyte API server")
-	quobyteConfigName := flag.String("configuration_name", "BASE", "Name of the volume configuration of new volumes")
-	quobyteAPIURL := flag.String("api", "http://localhost:7860", "URL to the API server(s) in the form http(s)://host[:port][,host:port] or SRV record name")
-	quobyteRegistry := flag.String("registry", "localhost:7861", "URL to the registry server(s) in the form of host[:port][,host:port] or SRV record name")
-	quobyteTenantId := flag.String("tenant_id", "no default", "Id of the Quobyte tenant in whose domain the operation takes place")
-
-	group := flag.String("group", "root", "Group to create the unix socket")
-	maxWaitTime := flag.Float64("max-wait-time", 30, "Maximimum wait time for filesystem checks to complete when a Volume is created before returning an error")
-	maxFSChecks := flag.Int("max-fs-checks", 5, "Maximimum number of filesystem checks when a Volume is created before returning an error")
 	showVersion := flag.Bool("version", false, "Shows version string")
+
 	flag.Parse()
 
+	maxFSChecks, _ := strconv.Atoi(os.Getenv("MAX_FS_CHECKS"))
+	maxWaitTime, _ := strconv.ParseFloat(os.Getenv("MAX_WAIT_TIME"), 64)
+	socketGroup := os.Getenv("SOCKET_GROUP")
+
+	quobyteAPIURL := os.Getenv("QUOBYTE_API_URL")
+	quobyteAPIPassword := os.Getenv("QUOBYTE_API_PASSWORD")
+	quobyteAPIUser := os.Getenv("QUOBYTE_API_USER")
+	quobyteMountPath := os.Getenv("QUOBYTE_MOUNT_PATH")
+	quobyteMountOptions := os.Getenv("QUOBYTE_MOUNT_OPTIONS")
+	quobyteRegistry := os.Getenv("QUOBYTE_REGISTRY")
+	quobyteTenantID := os.Getenv("QUOBYTE_TENANT_ID")
+	quobyteVolConfigName := os.Getenv("QUOBYTE_VOLUME_CONFIG_NAME")
```

Comment:
What about using some sane defaults here e.g. if a variable is not set?

Reply:
Sane defaults can be provided for most but not all params. Therefore the config file is needed anyways and cannot be omitted. I'm not sure providing defaults is helpful if the config file with some settings is required anyways. Am thinking about this...